### PR TITLE
[Improvement] Refactor Swin-Transformer

### PR DESCRIPTION
**Pull request by xvjiarui**
## Motivation

Refactor Swin-Transformer related code as https://github.com/open-mmlab/mmdetection/pull/5748

## BC-breaking (Optional)

We no longer support directly loading official weight [here](https://github.com/microsoft/Swin-Transformer), instead, we will use the conversion script in #783

### CODE REVIEW

File: mmseg/models/backbones/swin.py [hunk 1]
```diff
@@ -601,17 +549,22 @@ def __init__(self,
...
+        if isinstance(pretrained, str):
+            warnings.warn('DeprecationWarning: pretrained is deprecated, '
                           'please use "init_cfg" instead')
+            self.init_cfg = dict(type='Pretrained', checkpoint=pretrained)
+        elif pretrained is None:
+            self.init_cfg = init_cfg
         else:
             raise TypeError('pretrained must be a str or None')
 
+        super(SwinTransformer, self).__init__(init_cfg=init_cfg)
```
Comment (Junjun2016) [inline mmseg/models/backbones/swin.py:563]: Should move this line to L541?
Will this override self.init_cfg (override L557)?

File: mmseg/models/utils/embed.py [hunk 2]
```diff
@@ -1,100 +1,335 @@
...
+    """Image to Patch Embedding.
 
     We use a conv layer to implement PatchEmbed.
+
     Args:
         in_channels (int): The num of input channels. Default: 3
         embed_dims (int): The dimensions of embedding. Default: 768
-        conv_type (dict, optional): The config dict for conv layers type
-            selection. Default: None.
+        conv_type (str): The config dict for embedding
```
Comment (Junjun2016) [inline mmseg/models/utils/embed.py:92]: Not `config dict` (str).

File: mmseg/models/utils/embed.py [hunk 3]
```diff
@@ -1,100 +1,335 @@
...
+            conv layer type selection. Default: "Conv2d.
```
Comment (Junjun2016) [inline mmseg/models/utils/embed.py:93]: conv layer type selection. Default: "Conv2d".

**Comment by Junjun2016** [comment]
resolve some comments.

File: tests/test_models/test_utils/test_embed.py [hunk 4]
```diff
@@ -0,0 +1,461 @@
...
+
+
+def test_adaptive_padding():
+
+    for padding in ('same', 'corner'):
+        kernel_size = 16
+        stride = 16
+        dilation = 1
+        input = torch.rand(1, 1, 15, 17)
+        pool = AdaptivePadding(
```
Comment (Junjun2016) [inline tests/test_models/test_utils/test_embed.py:15]: `pool` --> `adap_pad`

File: mmseg/models/utils/embed.py [hunk 5]
```diff
@@ -1,100 +1,335 @@
...
+    """Applies padding to input (if needed) so that input can get fully covered
+    by filter you specified. It support two modes "same" and "corner". The
+    "same" mode is same with "SAME" padding mode in TensorFlow, pad zero around
+    input. The "corner"  mode would pad zero to bottom right.
+
+    Args:
+        kernel_size (int | tuple): Size of the kernel:
+        stride (int | tuple): Stride of the filter. Default: 1:
+        dilation (int | tuple): Spacing between kernel elements.
+            Default: 1
```
Comment (Junjun2016) [inline mmseg/models/utils/embed.py:22]:
```suggestion
            Default: 1.
```

File: mmseg/models/utils/embed.py [hunk 6]
```diff
@@ -1,100 +1,335 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+import math
+from typing import Sequence
+
+import torch.nn as nn
 import torch.nn.functional as F
 from mmcv.cnn import build_conv_layer, build_norm_layer
 from mmcv.runner.base_module import BaseModule
 from torch.nn.modules.utils import _pair as to_2tuple
```
Comment (Junjun2016) [inline mmseg/models/utils/embed.py:9]: https://github.com/pytorch/pytorch/commit/2503028ff5bab90a6c93687ce4e294815f3e243a

Comment (Junjun2016) [inline mmseg/models/utils/embed.py:9]:
```suggestion
from mmcv.utils import to_2tuple
```

**Review by Junjun2016 — APPROVED** [review]
LGTM, it seems that the code coverage should improve.

**Comment by xvjiarui** [comment]
Coverage improved. The rest should come from the loading checkpoint part.

**Review by Junjun2016 — APPROVED** [review]
LGTM and I will re-run the benchmark to double-check correctness.

**Review by Junjun2016 — APPROVED** [review]

**Pull request merged**